src/Balancing/Algorithm/RoundRobin.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Utopia\Balancing\Algorithm;
+
+use Utopia\Balancing\Algorithm;
+use Utopia\Balancing\Option;
+
+class RoundRobin extends Algorithm
+{
+    private int $index = 0;
+
+    /**
+     * @param Option[] $options
+     */
+    public function run(array $options): ?Option
```

Comment:
How do we tell what was the previous index? This class should be stateless. Maybe our process/request/script can pass the last index in the constructor or we can store it statically.

Reply:
Hmm this is no longer atomic as well. To make it atomic, we would need Swoole, which I don't think we want as part of this library at all. 

Maybe we can just use a static variable here, and for OPR proxy we write OPR-specific atomic round robin using Swoole?

Reply:
We can, but I'm not convinced we need to rewrite the entire thing in OPR. We can use a Swoole Table to share the index between workers.